### PR TITLE
Subtitles improvements

### PR DESCRIPTION
**Pull request by jaimeMF**
Basically: for each language we give a list containing all the formats available, and `YoutubeDL` is now in charge of selecting the language and formats to download.

This tries to address #4019 and #3804.

Here's a list of the extractor that need to be converted:
* [x] atresplayer
* [x] bbccouk
* [x] bliptv
* [x] ceskatelevize
* [x] crunchyroll
* [x] dailymotion
* [x] drtv
* [x] lynda
* [x] mit
* [x] mtv
* [x] npo
* [x] nrk
* [x] rai
* [x] ted
* [x] theplatform
* [x] viki
* [x] vimeo
* [x] walla
* [x] youtube

### CODE REVIEW

**Comment by jaimeMF** [comment]
In case you don't have any objection/suggestion, don't merge this. I would like to migrate all the extractors first, and since it changes how the subtitles are stored in the json output I want to talk to the mpv team so that they can update the youtube-dl support (if you know of any project that uses it, we may also want to contact them).

File: youtube_dl/YoutubeDL.py [hunk 1]
```diff
@@ -1586,6 +1655,18 @@ def list_thumbnails(self, info_dict):
             ['ID', 'width', 'height', 'URL'],
             [[t['id'], t.get('width', 'unknown'), t.get('height', 'unknown'), t['url']] for t in thumbnails]))
 
+    def list_subtitles(self, video_id, subtitles, name='subtitles'):
+        if not subtitles:
+            self.to_screen('%s has no %s' % (video_id, name))
+            return
+        header_line = 'Language    formats'
+        sub_lines = [
```
Comment (phihag) [inline youtube_dl/YoutubeDL.py]: Can we use `render_table` from utils here?

**Comment by phihag** [comment]
LGTM, feel free to merge when you think it's ready.

File: youtube_dl/YoutubeDL.py (hunk 1 above)
Comment (jaimeMF) [inline youtube_dl/YoutubeDL.py]: Yes, I forgot to change it, I wrote this part before you added the function.

Comment (jaimeMF) [inline youtube_dl/YoutubeDL.py]: Changed in edab9dbf4d00a7f76fbfd2df9ef4b205c88e47a8

**Comment by jaimeMF** [comment]
I have converted all the extractors, I'll probably add/clean some test and merge it this weekend.

**Pull request merged**